recipe_scrapers/joythebaker.py

```diff
@@ -0,0 +1,84 @@
+# mypy: allow-untyped-defs
+
+from __future__ import annotations
+
+from recipe_scrapers._exceptions import ElementNotFoundInHtml
+from recipe_scrapers._grouping_utils import IngredientGroup
+from recipe_scrapers._utils import get_minutes
+
+from ._abstract import AbstractScraper
+
+
+class JoyTheBaker(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "joythebaker.com"
+
+    def author(self):
+        return self.schema.author()
+
+    def title(self):
+        return self.schema.title()
+
+    def category(self):
+        return self.schema.category()
+
+    def total_time(self):
+        recipe_time = self.soup.find("span", {"class": "tasty-recipes-total-time"})
+        try:
+            return get_minutes(recipe_time)
+        except ElementNotFoundInHtml:
+            return None
+
+    def yields(self):
+        return self.schema.yields()
+
+    def image(self):
+        return self.schema.image()
+
+    def ingredients(self):
+        return self.schema.ingredients()
+
+    def ingredient_groups(self) -> list[IngredientGroup]:
+        ingredients = self.soup.find("div", {"class": "tasty-recipes-ingredients-body"})
+
+        categories = ingredients.find_all("p")
+        ingredient_groupings = ingredients.find_all("ul")
+
+        ingredient_groups = []
+
+        for index, group in enumerate(categories):
+            ingredient_list = ingredient_groupings[index].find_all("li")
+            ingredient_groups.append(
+                IngredientGroup(
+                    [ingredient.text for ingredient in ingredient_list], group.text
+                )
+            )
```

Comment:
Could we iterate the `ingredient_groupings` directly, without needing to query the `categories` paragraphs?

(not essential, but maybe a nice cleanup if possible -- and if the results stay correct)

Reply:
Ah, d'oh - perhaps not, because the `group.text` value is required.  I think I didn't spot that because I didn't immediately connect the `group` variable name to the `categories` collection it relates to.

Reply:
Something along these lines?
`for category, ingredient_group in zip(categories, ingredient_groupings):
            ingredient_groups.append(
                IngredientGroup(
                    [ingredient.text for ingredient in ingredient_group], category.text
                )
            )`

Reply:
That's neat, yep!